### PR TITLE
Disaster Counter tweaks

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

Makes Disaster Counters reset their current record after any explosion, no matter the severity.
Also makes the maptext update with the counter, and clamped the visual counter to a minimum of 0, since -1 doesn't make much sense.
*(Requested by @AffectedArc07)*

## Why It's Good For The Game

Better visually since the numbers actually update, and it'll be a lot more reliable for detecting explosions.

## Changelog
:cl:
tweak: Disaster Counters are now reset by all explosion severities.
fix: Fixed Disaster Counters showing -1 in their examine text.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/disaster_counter.dm [hunk 1]
```diff
@@ -30,16 +30,19 @@
 	SSpersistent_data.register(src)
 
 /obj/structure/disaster_counter/ex_act(severity)
-	if(severity < 2)
-		current_count = -1
-		persistent_save()
 	. = ..()
+	current_count = -1
+	persistent_save()
+	update_maptext()
```
Comment (AffectedArc07) [inline code/game/objects/structures/disaster_counter.dm:36]:
```suggestion
	current_count = -1
	persistent_save()
	update_maptext()
	. = ..()
```

Do it this way round incase it qdels in `..()`

**Review by Kyep — APPROVED** [review]

File: code/game/objects/structures/disaster_counter.dm [hunk 2]
```diff
@@ -30,16 +31,19 @@
...
 	. = ..()
 
 /obj/structure/disaster_counter/Destroy()
 	if(counter_id)
 		SSpersistent_data.registered_atoms -= src // Take us out the list
 	return ..()
 
+/obj/structure/disaster_counter/proc/update_maptext()
+	maptext = "<span class='maptext' style='text-align: center'>[max(current_count, 0)] / [record_count]</span>"
```
Comment (AffectedArc07) [inline code/game/objects/structures/disaster_counter.dm:45]:
```suggestion
	maptext = "<span class='maptext' style='text-align: center'>[max(current_count, 0)]/[record_count]</span>"
```
This spacing has an ingame effect. Its a string, not a mathematical operation.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/disaster_counter.dm (hunk 2 above)
Comment (SabreML) [reply code/game/objects/structures/disaster_counter.dm:45]: Yeah I realise, it just looked better to me with spacing between the numbers.
NSS Cyberiad 2021-03-01 200435
NSS Cyberiad 2021-03-01 200423

Comment (AffectedArc07) [reply code/game/objects/structures/disaster_counter.dm:45]: Try bumping the values to bigger numbers like 42 on each side and see how it looks

Comment (SabreML) [reply code/game/objects/structures/disaster_counter.dm:45]: NSS Cyberiad 2021-03-01 203009

It only really starts clipping out of the screen when it goes above five digits, but fair point.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**